harvester_e2e_tests/integration/test_hosts.py

```diff
@@ -398,72 +422,72 @@ def test_delete_vm_after_host_shutdown(
-    while endtime > datetime.now():
-        l_check = dict()
-        for vol_name, (code, data) in check.items():
-            if 200 != code:
-                fails.append((vol_name, f"Failed to delete\nStatus({code}): {data}"))
+        fails, check = [], dict()
+        for vol in spec.volumes:
+            vol_name = vol['volume']['persistentVolumeClaim']['claimName']
+            code, data = api_client.volumes.delete(vol_name)
+            if 200 == code:
+                check[vol_name] = (code, data)
             else:
-                code, data = api_client.volumes.get(vol_name)
-                if 404 != code:
-                    l_check[vol_name] = (code, data)
-        check = l_check
-        if not check:
-            break
-        sleep(5)
-    else:
-        for vol_name, (code, data) in check.items():
-            fails.append((vol_name, f"Failed to delete\nStatus({code}): {data}"))
+                fails[vol_name] = (code, data)
 
-    assert not fails, (
-        f"Failed to delete VM({focal_vm['name']})'s volumes with errors:\n"
-        "\n".join(f"Volume({n}): {r}" for n, r in fails)
-    )
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            for vol_name, (code, data) in check.copy().items():
+                code, data = api_client.volumes.get(vol_name)
+                if 404 == code:
+                    check.pop(vol_name)
+            if not check:
+                break
+            sleep(5)
+        else:
+            for vol_name, (code, data) in check.items():
+                fails.append((vol_name, f"Failed to delete\nStatus({code}): {data}"))
 
-    # teardown: power on the host
-    rc, out, err = host_state.power(node['id'], node_ip, on=True)
-    assert rc == 0, (f"Failed to PowerOn node {node['id']} with error({rc}):\n"
-                     f"stdout: {out}\n\nstderr: {err}")
-    sleep(host_state.delay)  # Wait for the node to disappear
-    endtime = datetime.now() + timedelta(seconds=wait_timeout)
-    while endtime > datetime.now():
-        _, metric = api_client.hosts.get_metrics(node['id'])
-        if not metric.get("metadata", {}).get("state", {}).get("error"):
-            break
-        sleep(5)
-    else:
-        raise AssertionError(
-            f"Node {node['id']} still unavailable after PowerOn script executed"
-            f", script path: {host_state.path}"
+        assert not fails, (
+            f"Failed to delete VM({focal_vm['name']})'s volumes with errors:\n"
+            "\n".join(f"Volume({n}): {r}" for n, r in fails)
         )
+    finally:
+        # teardown: power on the host
+        rc, out, err = host_state.power(node['id'], node_ip, on=True)
+        assert rc == 0, (f"Failed to PowerOn node {node['id']} with error({rc}):\n"
+                         f"stdout: {out}\n\nstderr: {err}")
+        sleep(host_state.delay)  # Wait for the node to disappear
```

Comment:
should be "appear"?
Or we can drop the comment directly cuz the code purpose is clear (for me, at least)